File: c#/src/Docmost.Sdk/Models/CreatePageDto.cs
namespace Docmost.Sdk.Models;

public class CreatePageDto
{
    [System.Text.Json.Serialization.JsonPropertyName("spaceId")]
    public string SpaceId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("title")]
    public string? Title { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("icon")]
    public string? Icon { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("parentPageId")]
    public string? ParentPageId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("content")]
    public string? Content { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("format")]
    public string Format { get; set; }

}

File: go/docmost/models/create_page_dto.go
package models

// CreatePageDto from OpenAPI.
type CreatePageDto struct {
	SpaceId string `json:"spaceId"`
	Title *string `json:"title,omitempty"`
	Icon *string `json:"icon,omitempty"`
	ParentPageId *string `json:"parentPageId,omitempty"`
	Content *string `json:"content,omitempty"`
	Format *string `json:"format,omitempty"`
}
